crates/viewer/re_renderer/src/video/player.rs

```diff
@@ -215,13 +215,12 @@ impl VideoPlayer {
         self.enqueue_samples(video_description, requested_sample_idx, video_buffers)?;
 
         // Grab best decoded frame for the requested PTS and discard all earlier frames to save memory.
-        if let Some(decoded_frame) = self
-            .sample_decoder
+        self.sample_decoder
             // Use the `requested_pts` which may be a bit higher than the PTS of the latest-at sample for `requested_pts`.
             // This is to hedge against not well-behaved decoders, that may produce PTS values that
             // don't show up in the input data (that in and on its own is a bug, but this makes it more robust)
-            .latest_decoded_frame_at_and_drop_earlier_frames(requested_pts)
-        {
+            .process_incoming_frames_and_drop_earlier_than(requested_pts);
+        if let Some(decoded_frame) = self.sample_decoder.oldest_available_frame() {
```

Comment:
I ran into borrow checker issues here, splitting the function was the most trivial way to resolve this